File: pwr_sem4/Aplikacje Mobilne/lab1/Zad2/src_Java/main/java/com/example/zad2_java/MainActivity.java
package com.example.zad2_java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private int pointsCPU;
    private int pointsPlayer;
    private int moveCPU;
    private int movePlayer;
    private TextView displayCPUMove;
    private TextView displayPlayerMove;
    private TextView displayCPUPoints;
    private TextView displayPlayerPoints;
    private Random rand;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        pointsCPU = 0;
        pointsPlayer = 0;
        displayCPUMove = findViewById(R.id.choiceCPU);
        displayCPUMove.setText("");
        displayPlayerMove = findViewById(R.id.choicePlayer);
        displayPlayerMove.setText("");
        displayCPUPoints = findViewById(R.id.pointsCPU);
        displayCPUPoints.setText("" + pointsCPU);
        displayPlayerPoints = findViewById(R.id.pointsPlayer);
        displayPlayerPoints.setText("" + pointsPlayer);
        rand = new Random();
    }
    private void roll() {
        moveCPU = rand.nextInt(3);
        switch (moveCPU) {
            case 0:
                displayCPUMove.setText("Kamień");
                break;
            case 1:
                displayCPUMove.setText("Papier");
                break;
            case 2:
                displayCPUMove.setText("Nożyce");
                break;
        }

    }
    private void fight() {
        switch (movePlayer) {
            case 0:
                displayPlayerMove.setText("Kamień");
                break;
            case 1:
                displayPlayerMove.setText("Papier");
                break;
            case 2:
                displayPlayerMove.setText("Nożyce");
                break;
        }
        if (moveCPU == movePlayer) {
            Toast.makeText(this, "Remis!", Toast.LENGTH_SHORT).show();
        } else if ((moveCPU == 0 && movePlayer == 1) || (moveCPU == 1 && movePlayer == 2) || (moveCPU == 2 && movePlayer == 0)) {
            pointsPlayer++;
            displayPlayerPoints.setText("" + pointsPlayer);
            Toast.makeText(this, "Wygrana!", Toast.LENGTH_SHORT).show();
        } else {
            pointsCPU++;
            displayCPUPoints.setText("" + pointsCPU);
            Toast.makeText(this, "Przegrana!", Toast.LENGTH_SHORT).show();
        }
    }
    public void rock(View view) {
        roll();
        movePlayer = 0;
        fight();
    }
    public void paper(View view) {
        roll();
        movePlayer = 1;
        fight();
    }
    public void scissors(View view) {
        roll();
        movePlayer = 2;
        fight();
    }
}

File: inzynier/pwr_sem4/Aplikacje Mobilne/lab1/Zad2/src_Kotlin/main/java/com/example/zad2/MainActivity.kt
package com.example.zad2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import android.widget.Toast
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    private var pointsCPU = 0
    private var pointsPlayer = 0
    private var moveCPU = 0     //0 = kamień, 1 = papier, 2 = nożyce
    private var movePlayer = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        pointsCPU = 0
        pointsPlayer = 0
        findViewById<TextView>(R.id.choiceCPU).text = ""
        findViewById<TextView>(R.id.choicePlayer).text = ""
        findViewById<TextView>(R.id.pointsCPU).text = "" + pointsCPU
        findViewById<TextView>(R.id.pointsPlayer).text = "" + pointsPlayer
    }
    private fun roll() {
        moveCPU = Random.nextInt(3)
        if (moveCPU == 0) {
            findViewById<TextView>(R.id.choiceCPU).text = "Kamień"
        }
        else if (moveCPU == 1) {
            findViewById<TextView>(R.id.choiceCPU).text = "Papier"
        }
        else {
            findViewById<TextView>(R.id.choiceCPU).text = "Nożyce"
        }
    }
    private fun fight() {
        if (movePlayer == 0) {
            findViewById<TextView>(R.id.choicePlayer).text = "Kamień"
        }
        else if (movePlayer == 1) {
            findViewById<TextView>(R.id.choicePlayer).text = "Papier"
        }
        else {
            findViewById<TextView>(R.id.choicePlayer).text = "Nożyce"
        }
        if (moveCPU == movePlayer) {
            Toast.makeText(this, "Remis!", Toast.LENGTH_SHORT).show()
        }
        else if ((moveCPU == 0 && movePlayer == 1) || (moveCPU == 1 && movePlayer == 2) || (moveCPU == 2 && movePlayer == 0)) {
            pointsPlayer++
            findViewById<TextView>(R.id.pointsPlayer).text = "" + pointsPlayer
            Toast.makeText(this, "Wygrana!", Toast.LENGTH_SHORT).show()
        }
        else {
            pointsCPU++
            findViewById<TextView>(R.id.pointsCPU).text = "" + pointsCPU
            Toast.makeText(this, "Przegrana!", Toast.LENGTH_SHORT).show()
        }
    }
    fun rock(view: View) {
        roll()
        movePlayer = 0
        fight()
    }
    fun paper(view: View) {
        roll()
        movePlayer = 1
        fight()
    }
    fun scissors(view: View) {
        roll()
        movePlayer = 2
        fight()
    }
}